Assets/VRTK/Scripts/VRTK_MoveInPlace.cs

```diff
@@ -0,0 +1,387 @@
+        public enum DirectionalMethod
+        {
+            Gaze,
+            DumbDecoupling,
+            SmartDecoupling
+        }
+
+        /// <summary>
+        /// If true, the left controller's trackpad will engage Move In Place.
+        /// </summary>
+        public bool LeftController
+        {
+            get { return leftController; }
+            set
+            {
+                leftController = value;
+                SetControllerListeners(controllerLeftHand);
+            }
+        }
+
+        /// <summary>
+        /// If true, the right controller's trackpad will engage Move In Place.
+        /// </summary>
+        public bool RightController
+        {
+            get { return rightController; }
+            set
+            {
+                rightController = value;
+                SetControllerListeners(controllerRightHand);
+            }
+        }
+
+        [Tooltip("If this is checked then the left controller touchpad will be enabled to move the play area. It can also be toggled at runtime.")]
+        [SerializeField]
+        private bool leftController = true;
+        [Tooltip("If this is checked then the right controller touchpad will be enabled to move the play area. It can also be toggled at runtime.")]
+        [SerializeField]
+        private bool rightController = true;
+
+        [Tooltip("Select which trackables are used to determine movement.")]
+        [SerializeField]
+        public ControlOptions controlOptions = ControlOptions.HeadsetAndControllers;
+
+        [SerializeField]
+        [Tooltip("Lower to decrease speed, raise to increase.")]
+        public float speedScale = 1;
+
+        [SerializeField]
+        [Tooltip("The max speed you can move in game units. (If 0 or less, max speed is uncapped)")]
+        public float maxSpeed = 4;
+
+        [SerializeField]
+        [Tooltip("How your movement direction will be determined.  The Gaze method tends to lead to the least motion sickness.  Smart decoupling is still a Work In Progress.")]
+        public DirectionalMethod dirMethod = DirectionalMethod.Gaze;
+
+        [SerializeField]
+        [Tooltip("If we're using Smart Decoupling, all trackables must be within this degree threshold to change direction.")]
+        public float smartDecoupleThreshold = 30f;
+
+        // The cap before we stop adding the delta to the movement list.  This will help regulate speed.
+        [SerializeField]
+        [Tooltip("The max amount of movement we wish to register in the virutal world.  Decreasing this will increase acceleration, and vice versa.")]
+        public float sensitivity = 0.02f;
+
+        // The maximum number of updates we should hold to process movements.  The higher the number, the slower the acceleration/deceleration & vice versa.
+        private int averagePeriod = 60;
+        private float cameraLastYRotation = 0f;
```

Comment:
cameraLastYRotation is set (and updated) but never used anywhere. It can be removed.